docs/source/realworld.rst

```diff
@@ -0,0 +1,89 @@
+SQL has been around for a long time, as a language for communicating
+with databases, like a communication protocol. More recently with the
+rise of *data* as a business function, or a domain in it's own right
+SQL has also become an invaluable tool for defining the *structure* of
+data and analysis - not just as a one off but as a form of
+`infrastructure as code`_.
+
+As *analytics* transitions from a profession of people doing one-offs,
+and moves to building stable and reuable pieces of analytics, more and
+more principles from software engineering are moving in the analytics
+space. One of the best articulations of this is written in the
+`viewpoint section of the docs for the open-source tool dbt`_. Two of
+the principles mentioned in that article are `quality assurance`_ and
+`modularity`_.
+
+Quality assurance
+-----------------
+
+The primary aim of `sqlfluff` as a project is in service of that first
+aim of `quality assurance`_. With larger and larger teams maintaining
+large bodies of SQL code, it becomes more and more important that the
+code is not just *valid* but also easily *comprehendable* by other users
+of the same codebase. One way to ensure readability is to enfore a
+`consistent style`_, and the tools used to to this are called `linters`_.
+
+Some famous `linters`_ which are well known in the software community are
+`flake8`_ and `jslint`_ (the former is used to lint the `sqlfluff` project
+itself).
+
+**Sqlfluff** aims to fill this space for SQL.
+
+Modularity
+----------
+
+SQL itself doesn't lend itself well to `modularity`_, so to introduce
+some flexibility and reusability it is often `templated`_. Typically
+this is done in the wild in one of the following ways:
+
+1. Using the limited inbuilt templating abilities of a programming
+   language directly. For example in python this would be using the
+   `format string syntax`_:
+
+   .. code-block:: python
+
+      "SELECT {foo} FROM {tbl}".format(foo="bar", tbl="mytable")
+
+   Which would evaluate to:
+
+   .. code-block:: sql
+
+      SELECT bar FROM mytable
+
+2. Using a dedicated templating library such as `jinja2`_. This allows
+   a lot more flexibility and more powerful expressions and macros. See
+   the :ref:`templateconfig` section for more detail on how this works.
+
+   - Often there are tools like `dbt`_ or `apache airflow`_ which allow
+     `templated`_ sql to be used directly, and they will implement a
+     library like `jinja2`_ under the hood themselves.
+
+
+All of these templating tools are great for `modularity`_ but they also
+mean that the SQL files themselves are no longer valid SQL code, because
+they now contain these configured *placeholder* values, inteded to
+improve modularity.
+
+Sqlfluff allows limited templating using both of the methods outlined
+above, to allow you to still lint those SQL files as part of your CI/CD
+pipeline (which is great 🙌), rather than waiting until you're in production
+(which is bad 🤦, and maybe too late).
+
```

Comment:
This is great! I suggest maybe one final paragraph, something like: 

>During the CI/CD pipeline, Sqlfluff needs additional info in order to interpret your templates as valid SQL code. You do so by providing dummy parameters in Sqlfluff configuration files. When substituted into the template, these values should evaluate to valid SQL (so Sqlfluff can check its style, formatting, and correctness), but the values _don't need to match actual values_ used in production.

Reply:
Nice. I've added another small section afterward to explain why we would actually *recommend* them being the *simplest* possible rather than even trying to emulate the actual values.